### PR TITLE
Use `ci4rail-base-image` as basis image of `ci4rail-devtools-image`

### PR DESCRIPTION
**Pull request by KairaMiu**
- Update license
- Derive `ci4rail-devtools-image` from `ci4rail-base-image`
- Move configs from local conf to image, if possible
- Add readme and layer dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# meta-ci4rail-bsp
```
Comment (KlausPopp) [inline README.md]: Adopt whole README to meta-ci4rail-devtools

File: conf/layer.conf [hunk 2]
```diff
@@ -11,3 +11,4 @@ BBFILE_PRIORITY_meta-ci4rail-devtools = "101"
 
 LAYERDEPENDS_meta-ci4rail-devtools = "core "
 LAYERSERIES_COMPAT_meta-ci4rail-devtools = "dunfell"
+LAYERDEPENDS_meta-ci4rail-devtools = "meta-ci4rail-bsp"
```
Comment (KlausPopp) [inline conf/layer.conf:14]: Shouldn't it be LAYERDEPENDS_meta-ci4rail-devtools += "meta-ci4rail-bsp" ?

**Review by KlausPopp — CHANGES_REQUESTED** [review]

File: conf/layer.conf (hunk 2 above)
Comment (KairaMiu) [reply conf/layer.conf:14]: No, see `LAYERDEPENDS` in [yocto reference manual](https://www.yoctoproject.org/docs/current/ref-manual/ref-manual.html)

File: README.md (hunk 1 above)
Comment (KairaMiu) [reply README.md]: ok

**Review by KlausPopp — APPROVED** [review]

**Pull request merged**